### ISSUE TITLE
Document `vm.derive`

### ISSUE THREAD

**Issue by onbjerg** [foundry-rs/book#452]
See https://github.com/foundry-rs/foundry/pull/2299

**Comment by Elyx0** [comment]
Am I missing something about the usage?

`source .env && forge script script/EOADeploy.s.sol:Deploy --fork-url http://localhost:8545 --broadcast`

```solidity
        uint256 deployerPrivateKey = vm.deriveKey(vm.envString("TESTNET_MNEMONIC"), 0);
 
        vm.startBroadcast(vm.addr(deployerPrivateKey));
```

```
Finding wallets for all the necessary addresses...

Transactions saved to: broadcast/EOADeploy.s.sol/80001/run-latest.json

Error:
No associated wallet for addresses: {0x1ba0f96b...4938cf70}. Unlocked wallets: []
error Command failed with exit code 1.
```

**Comment by onbjerg** [comment]
You still need to pass in the mnemonic separately to `forge script` so it knows the private key of the wallet when it has to sign transactions

**Comment by Elyx0** [comment]
It kinda defeats the purpose of self-aware scripting from env

**Comment by onbjerg** [comment]
This is not the place to discuss the feature itself, this is only for docs

**Comment by devanonon** [comment]
@Elyx0 have you started discussing it somewhere else? FYI, I am also working on a `remember` cheatcode which will allow you to actually add the private key to Foundry's wallet. I have not had the chance to finish it yet, but once I do I'll update docs to explain the intended usage.

**Comment by devanonon** [comment]
@Elyx0 `rememberKey` landed in #3047. I'll have docs in the book soon but you can play around with it once the next nightly lands in about 2 hours.
